lib/article_json/export/apple_news/elements/embed.rb

```diff
@@ -0,0 +1,100 @@
+module ArticleJSON
+  module Export
+    module AppleNews
+      module Elements
+        class Embed < Base
+          # Embed
+          # @return [Hash]
+          def export
+            {
+              role: role,
+              URL: source_url,
+              caption: caption
+            }.compact
+          end
+
+          private
+
+          def role
+            @role ||=
+              case embed_type
+              when :youtube_video, :vimeo_video, :dailymotion_video
+                :embedwebvideo
+              when :facebook_video
+                :facebook_post
+              when :tweet
+                :tweet
+              when :slideshare
+                nil
+              when :soundcloud
+                nil
+              else
+                nil
+              end
+          end
+
+          def source_url
+            case embed_type
+            when :youtube_video
+              build_embeded_youtube_url
+            when :vimeo_video
+              build_embeded_vimeo_url
+            when :dailymotion_video
+              build_embeded_vimeo_url
+            when :facebook_video
+              build_facebook_video_url
+            when :tweet
+              build_twitter_url
+            when :slideshare
+              nil
+            when :soundcloud
+              nil
+            else
+              nil
+            end
+          end
+
+          def caption
+            return nil if role.nil?
+
+            @element.caption.first&.content
+          end
+
+          def build_embeded_youtube_url
+            "https://www.youtube.com/embed/#{embed_id}"
+          end
+
+          def build_embeded_vimeo_url
+            "https://player.vimeo.com/video/#{embed_id}"
+          end
+
+          def build_embeded_dailymotion_url
+            "https://geo.dailymotion.com/player.html?video=#{embed_id}"
+          end
+
+          def build_facebook_video_url
+            embed_params = embed_id.to_s.split("/")
+            username = embed_params[0]
+            embed_id = embed_params[1]
+            "https://www.facebook.com/#{username}/videos/#{embed_id}"
+          end
+
+          def build_twitter_url
+            embed_params = embed_id.to_s.split("/")
```

Comment:
Not a big deal, but could be simplified like this:
```ruby
username, embed_id = embed_id.to_s.split("/").take(2)
```

Reply:
The suggestion should work in theory, but the variable cannot be reused it seems.

Amened without the take `method`.